gluster_exporter/metric_brick.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"fmt"
+	"encoding/json"
+	"net"
+	"net/http"
+	"os"
+	"io/ioutil"
+	"syscall"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	brick_labels = []string{
+		"host",
+		"id",
+		"brick_path",
+		"volume",
+	}
+
+	glusterBrickUsedCapacity = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "gluster",
+			Name:      "brick_used_capacity",
+			Help:      "Used capacity of gluster bricks",
+		},
+		brick_labels,
+	)
+
+	glusterBrickFreeCapacity = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "gluster",
+			Name:      "brick_free_capacity",
+			Help:      "Free capacity of gluster bricks",
+		},
+		brick_labels,
+	)
+
+	glusterBrickAllCapacity = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "gluster",
+			Name:      "brick_all_capacity",
+			Help:      "Total capacity of gluster bricks",
+		},
+		brick_labels,
+	)
```

Comment:
Add one more metric `brick_effective_used_capacity`, which can calculate used capacity based on sub volume type as described here. http://aravindavk.in/blog/monitoring-glusterfs-volume-utilization/

Reply:
Also add metrics related to inodes

Reply:
Regarding inodes, you mean getting details from iostats and then maintain as part bricks?

Reply:
No from the same statvfs output. (`df -i <mount-point>`)

        disk.InodesAll = uint64(fs.Files)
	disk.InodesFree = uint64(fs.Ffree)
	disk.InodesUsed = disk.InodesAll - disk.InodesFree

Reply:
brick_effective_used_capacity, may be will do as part of separate PR